app/src/main/java/com/github/h3lp3rs/h3lp/HelpPageActivity.kt

```diff
@@ -125,15 +131,44 @@ class HelpPageActivity : AppCompatActivity(), CoroutineScope by MainScope() {
         }
     }
 
-
     /** Starts the activity by sending intent */
     private fun goToActivity(ActivityName: Class<*>?) {
         val intent = Intent(this, ActivityName)
         startActivity(intent)
     }
 
-    fun goToMainPage(view: View) {
-        goToActivity(MainPageActivity::class.java)
+    /**
+     * Triggered when clicked on the cross
+     */
+    fun refuseHelp(view: View) {
+        goToMainPage()
     }
 
+    /**
+     * Triggered when the helper accepts to help
+     * @param view The view of the button
+     */
+    fun acceptHelp(view: View) {
+        if(helpId == null) {
+            goToMainPage()
+            return
+        }
+        databaseOf(EMERGENCIES).getObject(helpId!!, EmergencyInformation::class.java).thenApply {
+            // Add the helper to the list of helpers
+            val me = Helper(userUid!!, currentLat, currentLong)
+            val helpers = ArrayList<Helper>()
+            if(!helpers.contains(me)) {
+                helpers.add(me)
+            }
+            // Stop listening to other emergencies
+            databaseOf(NEW_EMERGENCIES).clearAllListeners()
```

Comment:
A discussion about that could be a good idea. 

We could have a case where an helper is closer to another emergency that which he was going and will not be notified